File: Gun/Assets/Scripts/Mechanics/FadeInOutBehaviour.js
#pragma strict
var colorStart : Color;
var colorEnd : Color;
var duration : float = 1.0;
var t : float;
var renderoption : Renderer;
static var playfadeIn : boolean;
static var playfadeOut: boolean;

function Start () {
	colorStart = renderoption.material.color;
	colorEnd = Color(colorStart.r, colorStart.g, colorStart.b, 0);
	//renderoption.material.color.a = 0;
	playfadeIn = false;
	playfadeOut = false;
}

 function Update () {
 	if(playfadeIn == true){
 		callFadeIn();
 		playfadeIn = false;
 	}
 	if(playfadeOut == true){
 		callFadeOut();
 		playfadeOut = false;
 	}

}

function callFadeIn(){

	fadeIn();
}

function callFadeOut(){
	fadeOut();

}

function fadeIn(){
	for(t = 0; t < duration; t+= Time.deltaTime){

	renderoption.material.color = Color.Lerp(colorStart, colorEnd, t/duration);
	yield;

	}
}

function fadeOut(){
	for(t = 0; t < duration; t+= Time.deltaTime){

	renderoption.material.color = Color.Lerp(colorEnd, colorStart, t/duration);
	yield;

	}
}

File: Assets/Scripts/Mechanics/FadeInOutBehaviour.cs
using UnityEngine;
using System.Collections;

public class fadeInOutBehaviour : MonoBehaviour {
    public Color colorStart;
    public Color colorEnd;
    public Renderer renderoption;
    public float duration = 1.0f;
    public float t = 1.0f;
    public bool  playfadeIn = false;
    public bool  playfadeOut = false;
    public scenesManager ScenesManager;

    void Start (){
	    colorStart = renderoption.material.color;
        colorEnd = new Color(colorStart.r, colorStart.g, colorStart.b, 0);
        ScenesManager = GameObject.Find("SceneManager").GetComponent<scenesManager>();
        ScenesManager.loadCurrentFade();
    }

    void Update (){
 	    if(playfadeIn == true){
            StartCoroutine(fadeIn());
            playfadeIn = false;
 	    }
 	    if(playfadeOut == true){
            StartCoroutine(fadeOut());
            playfadeOut = false;
 	    }
    }

    IEnumerator fadeIn (){
        for (t = 0; t < duration; t+= Time.deltaTime){
	        renderoption.material.color = Color.Lerp(colorStart, colorEnd, t/duration);
	        yield return 0;
	    }
    }

    IEnumerator fadeOut (){
	    for(t = 0; t < duration; t+= Time.deltaTime){
	        renderoption.material.color = Color.Lerp(colorEnd, colorStart, t/duration);
            yield return 0;
	    }
    }
}